File: model/src/main/java/com/testerum/model/step/operation/response/CheckComposedStepDefUpdateCompatibilityResponse.kt
package com.testerum.model.step.operation.response

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.testerum.model.infrastructure.path.Path

data class CheckComposedStepDefUpdateCompatibilityResponse @JsonCreator constructor(
        @JsonProperty("isCompatible") val isCompatible: Boolean = true,
        @JsonProperty("isUniqueStepPattern") val isUniqueStepPattern: Boolean = true,
        @JsonProperty("pathsForAffectedTests") val pathsForAffectedTests: List<Path> = emptyList(),
        @JsonProperty("pathsForDirectAffectedSteps") val pathsForDirectAffectedSteps: List<Path> = emptyList(),
        @JsonProperty("pathsForTransitiveAffectedSteps") val pathsForTransitiveAffectedSteps: List<Path> = emptyList()
) {
    companion object {
        val FULLY_COMPATIBLE = CheckComposedStepDefUpdateCompatibilityResponse(
                isCompatible = true,
                isUniqueStepPattern = true
        )
    }
}

File: old/web/web-frontend/src/app/model/step/operation/CheckComposedStepDefUpdateCompatibilityResponse.ts
import {Path} from "../../infrastructure/path/path.model";
import {Serializable} from "../../infrastructure/serializable.model";

export class CheckComposedStepDefUpdateCompatibilityResponse implements Serializable<CheckComposedStepDefUpdateCompatibilityResponse>{

    isCompatible: boolean;
    isUniqueStepPattern: boolean;
    pathsForAffectedTests: Array<Path> = [];
    pathsForDirectAffectedSteps: Array<Path> = [];
    pathsForTransitiveAffectedSteps: Array<Path> = [];

    deserialize(input: Object): CheckComposedStepDefUpdateCompatibilityResponse {
        this.isCompatible = input["isCompatible"];
        this.isUniqueStepPattern = input["isUniqueStepPattern"];

        for (let pathsForAffectedTest of (input['pathsForAffectedTests']) || []) {
            this.pathsForAffectedTests.push(Path.deserialize(pathsForAffectedTest));
        }

        for (let pathsForDirectAffectedStep of (input['pathsForDirectAffectedSteps']) || []) {
            this.pathsForDirectAffectedSteps.push(Path.deserialize(pathsForDirectAffectedStep));
        }

        for (let pathsForTransitiveAffectedStep of (input['pathsForTransitiveAffectedSteps']) || []) {
            this.pathsForTransitiveAffectedSteps.push(Path.deserialize(pathsForTransitiveAffectedStep));
        }

        return this;
    }

    serialize(): string {
        throw new Error("Not implemented method");
    }
}
